packages/browser-wallet/src/popup/pages/Setup/ChooseNetwork.tsx

```diff
@@ -3,15 +3,28 @@ import React from 'react';
 import { useNavigate } from 'react-router-dom';
 import Button from '@popup/shared/Button';
 import { absoluteRoutes } from '@popup/constants/routes';
-import { useSetAtom } from 'jotai';
+import { useSetAtom, useAtomValue } from 'jotai';
 import { useTranslation } from 'react-i18next';
+import { seedPhraseAtom } from '@popup/state';
 import { networkConfigurationAtom } from '@popup/store/settings';
-import { mainnet, testnet } from '../NetworkSettings/NetworkSettings';
+import { setupRoutes } from './routes';
+// TODO Remove stagenet
+import { mainnet, testnet, stagenet } from '../NetworkSettings/NetworkSettings';
 
 export function ChooseNetwork() {
     const navigate = useNavigate();
     const { t } = useTranslation('setup');
     const setNetworkConfiguration = useSetAtom(networkConfigurationAtom);
+    // TODO Make a better way to check if we are recovering
```

Comment:
Yes, this should be more explicit. If we save it in storage, then we would also be able (in the future) to redirect the UI to the recovery screen if the user opens the wallet after having closed mid-recovery. Or show them the result screen if it completed while being closed etc.

Reply:
I made a specific atom for it. We can put it into storage when we actually use it.